importer/main.py

```diff
@@ -475,6 +475,77 @@ def care_team_extras(
     return payload_string
 
 
+# This function takes in a payload_obj, a position and a tracker
+# It deletes the object at the given position in the resource->characteristic
+# It then returns the updated payload_obj and a new tracker for the remaining objects
+def delete_missing_obj_from_payload(payload_obj, position, tracker):
+    del payload_obj["resource"]["characteristic"][position]
+    return payload_obj, tracker-1
+
+
+# custom extras for product import
+def product_extras(resource, payload_string):
+    payload_obj = json.loads(payload_string)
+    product_name = resource[0]
+    tracker = 0
+
+    try:
+        (_, active, *_, previous_id, is_attractive_item, availability, condition, appropriate_usage,
+         accountability_period, image_source_url) = resource
+    except ValueError:
+        logging.error("Skipping: " + product_name + " : Because of missing columns ")
+        active = previous_id = is_attractive_item = availability = condition = appropriate_usage = \
+            accountability_period = image_source_url = "missing_column"
+
+    if active and active != "missing_column":
+        payload_obj["resource"]["active"] = active
+    else:
+        del payload_obj["resource"]["active"]
+
+    if previous_id and previous_id != "missing_column":
+        payload_obj["resource"]["identifier"][1]["value"] = previous_id
+    else:
+        del payload_obj["resource"]["identifier"][1]
```

Comment:
how do you know this is item 2, i.e. in slot index 1? how do you know the list contains more than 1 item?

Reply:
Because this is from the json_payload [here](importer/json_payloads/product_group_payload.json) which we built, there is always an official and a secondary id (this is specifically for group resources in this case)

Reply:
k need to encode these special numbers in a config structure then, a constant dict might be reasonable for that for now, like

```
GROUP_INDEX_MAPPING = {
"secondary_id_index": 1
}
```

then here instead of 1 you can say `GROUP_INDEX_MAPPING["secondary_id_index"]` so it's explicit what the index reps

Reply:
https://github.com/onaio/fhir-tooling/pull/179/commits/32bf3de2488250a6df960939bd30505dec3b3f79

Seems a bit verbose what do you think?

Reply:
I think this is better and clearer, and this way if the underlying structure changes you can change the map without having to change the rest of the code